### PR TITLE
t/6: Basic DOM utils.

### LINKED ISSUE #6: DOM utils
Moved from https://github.com/ckeditor/ckeditor5-core/issues/157

### PR DESCRIPTION
**Pull request by pjasiun**
Fixes #6.

### CODE REVIEW

File: src/dom/createelement.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ *		createElement( document, 'p', { class: 'foo' } ); // <p class="foo">
+ *		createElement( document, 'p', null, 'foo' ); // <p>foo</p>
+ *		createElement( document, 'p', null, [ 'foo', createElement( document, 'img' ) ] ); // <p>foo<img></p>
+ *
+ * @param {Document} doc Document used to create element.
+ * @param {String} name Name of the element.
+ * @param {Object} attributes Object keys will become attributes keys and object values will became attributes values.
+ * @param {Node|String|Array.<Node|String>} children Child or array of children. Strings will be automatically turned
+ * into Text nodes.
+ * @returns {Element} Created element.
```
Comment (Reinmar) [inline src/dom/createelement.js]: Missing `@method utils.dom.foo` in all files

File: src/dom/createelement.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import isString from '../lib/lodash/isString.js';
+import utils from '../utils.js';
+
+/**
+ * Create element with attributes and children.
```
Comment (Reinmar) [inline src/dom/createelement.js]: `s/Create/Creates`

File: src/dom/createelement.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * @returns {Element} Created element.
+ */
+export default function createElement( doc, name, attributes = {}, children = [] ) {
+	const element = doc.createElement( name );
+
+	for ( let key in attributes ) {
+		element.setAttribute( key, attributes[ key ] );
+	}
+
+	if ( isString( children ) || !utils.isIterable( children ) ) {
```
Comment (Reinmar) [inline src/dom/createelement.js]: Any reason why not `typeof c == 'string'`?

File: src/dom/insertat.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Insert node to the parent at given index.
```
Comment (Reinmar) [inline src/dom/insertat.js]: `s/Insert/Inserts/`

File: src/dom/remove.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Remove given node from parent.
```
Comment (Reinmar) [inline src/dom/remove.js]: `s/Remove/Removes/`

File: src/dom/indexof.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Returns index of the node in the parent element.
+ *
+ * @returns {Number} Index of the node in the parent element. Returns 0 if node has no parent.
```
Comment (Reinmar) [inline src/dom/indexof.js]: Missing `@param`.

File: src/dom/createelement.js (hunk 3 above)
Comment (pjasiun) [inline src/dom/createelement.js]: Any reason why not `isString`? ;) I do not like spending any second thinking how to check particular type in JS and really like being able to simply write `isString`, `isPlainObject`, `isIterable`. Also note that we already use `isString` in the engine (in schema).

File: src/dom/createelement.js (hunk 1 above)
Comment (pjasiun) [inline src/dom/createelement.js]: Fixed.

File: src/dom/createelement.js (hunk 3 above)
Comment (Reinmar) [inline src/dom/createelement.js]: Well, you didn't spend any more time, but I did and so you did ;). And if we agree to such convention we're not consistent yet so it deserves an issue.

Comment (pjasiun) [inline src/dom/createelement.js]: Agree, will you create a ticket?

**Comment by Reinmar** [comment]
Reviewed, and it's OK. I haven't run tests though – make sure it's green ;).

**Pull request merged**